src/integration.test.ts

```diff
@@ -531,3 +598,45 @@ describe(`Integration test for portal ${portal}`, () => {
     });
   });
 });
+
+/**
+ * Runs the full etag test on the given skylinks.
+ *
+ * @param skylink1 - The first skylink.
+ * @param skylink2 - The second skylink.
+ */
+export async function etagTestSkylinks(skylink1: string, skylink2: string): Promise<void> {
+  // The skylinks should differ.
+  expect(skylink1).not.toEqual(skylink2);
+
+  // Download the files.
+  let [url1, url2] = await Promise.all([client.getSkylinkUrl(skylink1), client.getSkylinkUrl(skylink2)]);
+  const [response1, response2] = await Promise.all([
+    // @ts-expect-error Calling a private method.
+    client.getFileContentRequest(url1),
+    // @ts-expect-error Calling a private method.
+    client.getFileContentRequest(url2),
+  ]);
+
+  // Get the etags.
+  const [etag1, etag2] = [response1.headers["etag"], response2.headers["etag"]];
+  expect(etag1).toBeTruthy();
+  expect(etag2).toBeTruthy();
+
+  // The etags should differ.
+  expect(etag1).not.toEqual(etag2);
+
+  // Download the files using nocache.
+  [url1, url2] = [`${url1}?nocache=true`, `${url2}?nocache=true`];
```

Comment:
you can also use `nocache` request header if that'd be easier

Reply:
Thanks for the suggestion, I didn't know that. We don't expose setting custom headers anymore, though.